Social messages backend

evandvance opened the pull request:
# What issue(s) does this PR address?

These changes don't have defined tickets specifically, but are a sans-websocket solution to the following tickets.. The tickets that were made for changes similar to these were referencing using WebSockets, which, to the best of my knowledge, is still being pursued. These changes are walking before we run. 

SENDME-68
SENDME-69
SENDME-71

This PR aims to bring the functionality of fetching the data for conversations a logged-in user is a part of from the backend. It also adds the functionality of getting the messages in that conversation and displaying them. It also allows a user to send new messages to the backend and uses a long polling strategy to keep the messages persisted updated on the backend. 

# What steps do reviewers need to take to review the PR?

1. Go To /conversations
    This action should seed the database with some fake users, conversations, and messages
2. Click the conversation with bob
    This action will redirect you to /conversations/$conversation_id
3. Send a message to Bob
4. Exit the Chat
5. Go back in
6. Observe the message is persisted

# Screenshots (if made UI changes)
No UI changes

# Note any new or changed endpoints.
/api/conversations
/conversations

# Known Remaining issues

- There is an odd display behavior with the names. A PR is being worked on to change the user shape. That will break this change. Once that PR goes in @evandvance is going to fix the issues with the users in this pr.
- The read behavior is not functioning correctly. It is currently stored in an association table, but it might need its modeling to behave the desired way without being super hacky. 

## Future goals

- Add frontend functionality for creating conversations 
- Implement WebSockets

# Checklist
- [x] Noted issue
- [x] Documented steps
- [x] Included Screenshots
- [x] OpenAPI Spec up-to-date (To the best of my knowledge)
- [x] Routes files up-to-date (To the best of my knowledge)
- [ ] Tests Passing
- [ ] Features Tested
- [ ] Code reviewed by Frontend/Backend leads
- [ ] Code reviewed by Andy

abbievance committed:
Initial backend connection

I know some things are buggy here, this was an initial test.

abbievance committed:
Dark mode ui fixes

WeslBaker committed:
Added essential Conversation and Message models

WeslBaker committed:
Added essential Conversation and Message models

WeslBaker committed:
Added essential Conversation and Message models

WeslBaker committed:
Added essential Conversation and Message models

WeslBaker committed:
Fixed ruff errors

WeslBaker committed:
Fixed ruff errors

WeslBaker committed:
Fixed ruff errors

abbievance committed:
Seed dummy data in the database

abbievance committed:
Delete option to delete from db

evandvance committed:
Conversation Restrucutre, Relational Mapping Stab 1

It was requested that this commit be pushed up so that the leaders could see where we are at

evandvance committed:
Add code that passes the smoke test for conversation endpoint

evandvance committed:
Current state of conversation after some pair programming

I am sorry for what I have done. Only I can bear the shame.

evandvance committed:
Fix relationships and add seed endpoint for testing

evandvance committed:
Made seeding messages, get all conversations, and get messages in conversation endpoints function

evandvance committed:
Add missed operation id

Commit:
Nulling

Placed null values under the messages and conversations until proper communication is created through identical data models both being pulled and being shown

Commit:
Modeling

evandvance committed:
Add useeffect to show working conversation displays

Commit:
More Modeling

Commit:
Backend Message Modeling

Commit:
Remove list

evandvance committed:
Revert schema changes

evandvance committed:
Make user conversations function for gets and display correctly on frontend

evandvance committed:
Make changes pass just pr

evandvance committed:
Update frontend to better handle future profile picture support

evandvance committed:
Add endpoints to create messages and conversations

evandvance committed:
Just openapi

evandvance committed:
Add ability to send message to conversation

evandvance committed:
Handle no message content

evandvance committed:
add hit enter to send

evandvance committed:
update conversation time on change

evandvance committed:
Use spinner in different spot, remove unneeded bloat

evandvance committed:
Remove useRef

evandvance committed:
Fix bad naming on routes

Commit:
Fixed Bug

Fixed bug that made the username have a ")" at the end of it in messages

abbievance committed:
Fix dark mode colors

abbievance committed:
biome fix

WeslBaker reviewed (approved):
Looks like we're headed for a merge conflict with the profiles branch, but I'm sure that'll be fine...

tuckerbrown5 committed:
Fixed ui

evandvance committed:
Fix formatting issues

evandvance committed:
Remove childless component

tuckerbrown5 committed:
some more fixing ui on this

tuckerbrown5 committed:
fixed biome issue?

tuckerbrown5 committed:
decreased space between bob and nav bar.

evandvance committed:
Profile static UI & User Creation (#35)

* Bare Minimum Structure for Profile

* Removed extra imports

* Biome Fix

* Biome Fixes

* Profile background component (#27)

* commit

* Biome Fixes

* Biome Fix

---------

Co-authored-by: Priscilla Harris <89226780+PHarris2002@users.noreply.github.com>

* Fix Tiny CSS Structure

* info created

* Biome Fix

* Add Static Tags Component (#31)

* Tags Component Progress

* Biome Fixes

* Add Profile Past Projects Component (#32)

* Past Projects Progress

* Biome Fix

* Profile Header Component (#33)

* Update header.tsx

* Update

* commit

* update header

* update header

* Biome Fix

---------

Co-authored-by: LukeNoles <89222532+LukeNoles@users.noreply.github.com>

* Merge and Biome Fix

* Tweak Some CSS

* Fix Header Placement (tacky)

* User creation (#34)

* Add User Router To Backend

* Update Opportunity router for better naming

* Add missing user feilds to db

* Add user Signup Form To frontend

* Run Autoformatter, Run just openapi

* Add redirect to sign-up on signin process

* Rework form to use formdata

* Add handling of duplicate emails

* Run Autoformatter, Fix pyright error

* Fix formdata errors

* Move Sign Up to Auth Route, Regenerate Route File

* Add feedback to tell user they are logged in

* Add darkmode and more consistent styling to login page

* Change signup to use placeholder instead of label

* Add type ignores

Adopted a new technology with stricter typing. Sendgrid (as far as I am aware) does not provide the support for typing. Could be wrong, but not a rabit worth chasing.

* Remove display name

* Add user socials to database models

* Add EmailStr type, Add socials to create user

* Add user socials to create user endpoint

* Make feilds none by default

* Add better form control to signup form

* Add tooltip to location feild

* Fix backend to be current understanding of shape of data

* Missed a field

* Add Desired signup flow

* Make code pass just pr

* Add login link to duplicate email message

Styles need to be added later.

* Update callback to use async/await for better readability

Ironic.

* Add youtube

* Add Edit Button + Biome Fix

* input hrefs + biome fix

* biome fix

* Making Page Editable + Biome Fix

* Update schema.ts

* Made social media icons optional and changed tag add button

* Update info.tsx

rename twitter-x to x

* Update info.tsx

---------

Co-authored-by: Priscilla Harris <89226780+PHarris2002@users.noreply.github.com>
Co-authored-by: LukeNoles <89222532+LukeNoles@users.noreply.github.com>
Co-authored-by: PearceWalker <100614334+PearceWalker@users.noreply.github.com>
Co-authored-by: Kenan Casey <kenancasey@gmail.com>
Co-authored-by: brigittet <89213142+brigittet@users.noreply.github.com>

evandvance committed:
Rename Display name to firstname on frontend

evandvance committed:
Rename display name on backend

evandvance committed:
Merge branch 'main' into social-messages-backend
